src/suspendplugin.c

```diff
@@ -266,6 +276,40 @@ suspend_handle_display_on_off_iface(
     }
 
     nlmsg_free(msg);
+
+    char cmd[512];
+    int cmd_len = 0;
+    struct ifreq ifr;
+    android_wifi_priv_cmd priv_cmd;
+
+    int ret;
+    int ioctl_sock;
+
+    ioctl_sock = socket(PF_INET, SOCK_DGRAM, 0);
+
+    snprintf(cmd, 512, "SETSUSPENDMODE %d", on_off);
+    cmd_len = strlen(cmd);
```

Comment:
snprintf returns the length, there's no need to calculate it again.

Reply:
It can also return negative value on errors, or "I would have written N bytes if there were that much space given" -> it can't be used as-is without sanity checking.

Reply:
what should the plugin do in case a negative value is returned?

Reply:
oh, come on, 512 bytes is more than enough for `"SETSUSPENDMODE %d"` :)

Reply:
so, ok to merge now?

Reply:
I would say yes :)